python/paddle/v2/framework/tests/op_test_util.py

```diff
@@ -27,17 +27,32 @@ def test_all(self):
                 places.append(core.GPUPlace(0))
 
             for place in places:
-                for in_name in Operator.get_op_input_names(self.type):
+                for ins in Operator.get_op_inputs(self.type):
+                    in_name = ins[0]
+                    in_dup = ins[1]
                     if hasattr(self, "inputs") and in_name in self.inputs:
-                        kwargs[in_name] = in_name
-                        var = scope.new_var(in_name).get_tensor()
-                        arr = self.inputs[in_name]
-                        var.set_dims(arr.shape)
-                        var.set(arr, place)
+                        kwargs[in_name] = []
+                        if in_dup:
+                            arrays = self.inputs[in_name]
+                            for index, arr in enumerate(arrays):
+                                var = scope.new_var(in_name + str(index))
+                                tensor = var.get_tensor()
+                                tensor.set_dims(arr.shape)
+                                tensor.set(arr, place)
+                                kwargs[in_name].append(in_name + str(index))
+                        else:
+                            var = scope.new_var(in_name)
+                            tensor = var.get_tensor()
+                            arr = self.inputs[in_name]
+                            tensor.set_dims(arr.shape)
+                            tensor.set(arr, place)
+                            kwargs[in_name].append(in_name)
                     else:
                         kwargs[in_name] = "@EMPTY@"
 
-                for out_name in Operator.get_op_output_names(self.type):
+                for outs in Operator.get_op_outputs(self.type):
```

Comment:
Maybe the multiple output also need the same logic with multiple inputs. And how about take this feature in an independent PR. , so that other Ops could also use this feature ASAP.

Reply:
Yes, the multiple output also need the same logic. I will split this pr into several small prs.